core/stateful_precompile_test.go

```diff
@@ -281,3 +283,300 @@ func TestContractDeployerAllowListRun(t *testing.T) {
 		})
 	}
 }
+
+func TestContractNativeMinterRun(t *testing.T) {
+	type test struct {
+		caller         common.Address
+		precompileAddr common.Address
+		input          func() []byte
+		suppliedGas    uint64
+		readOnly       bool
+
+		setupState  func(state *state.StateDB)
+		expectedRes []byte
+		expectedErr string
+
+		assertState func(t *testing.T, state *state.StateDB)
+	}
+
+	adminAddr := common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC")
+	noRoleAddr := common.HexToAddress("0xF60C45c607D0f41687c94C314d300f483661E13a")
+
+	for name, test := range map[string]test{
+		"noRole tries minting": {
+			caller:         noRoleAddr,
+			precompileAddr: precompile.ContractNativeMinterAddress,
+			input: func() []byte {
+				input, err := precompile.PackMintInput(noRoleAddr, common.Big1)
+				if err != nil {
+					panic(err)
+				}
+				return input
+			},
+			suppliedGas: precompile.MintGasCost,
+			readOnly:    false,
+			setupState: func(state *state.StateDB) {
+				precompile.SetContractNativeMinterStatus(state, adminAddr, precompile.AllowListAdmin)
+			},
+			expectedErr: precompile.ErrCannotMint.Error(),
+			assertState: func(t *testing.T, state *state.StateDB) {
+				res := precompile.GetContractNativeMinterStatus(state, adminAddr)
+				assert.Equal(t, precompile.AllowListAdmin, res)
+
+				res = precompile.GetContractNativeMinterStatus(state, noRoleAddr)
+				assert.Equal(t, precompile.AllowListAdmin, res)
+			},
+		},
+		"admin tries minting": {
+			caller:         adminAddr,
+			precompileAddr: precompile.ContractNativeMinterAddress,
+			input: func() []byte {
+				input, err := precompile.PackMintInput(adminAddr, common.Big1)
+				if err != nil {
+					panic(err)
+				}
+				return input
+			},
+			suppliedGas: precompile.MintGasCost,
+			readOnly:    false,
+			setupState: func(state *state.StateDB) {
+				precompile.SetContractNativeMinterStatus(state, adminAddr, precompile.AllowListAdmin)
+			},
+			expectedErr: precompile.ErrCannotMint.Error(),
+			assertState: func(t *testing.T, state *state.StateDB) {
+				res := precompile.GetContractNativeMinterStatus(state, adminAddr)
+				assert.Equal(t, precompile.AllowListAdmin, res)
+
+				res = precompile.GetContractNativeMinterStatus(state, noRoleAddr)
+				assert.Equal(t, precompile.AllowListAdmin, res)
+			},
+		},
+		"mint with readOnly enabled": {
+			caller:         noRoleAddr,
+			precompileAddr: precompile.ContractNativeMinterAddress,
+			input: func() []byte {
+				input, err := precompile.PackModifyAllowList(noRoleAddr, precompile.AllowListEnabled)
+				if err != nil {
+					panic(err)
+				}
+				return input
+			},
+			suppliedGas: precompile.ModifyAllowListGasCost,
+			readOnly:    true,
+			setupState: func(state *state.StateDB) {
+				precompile.SetContractNativeMinterStatus(state, adminAddr, precompile.AllowListAdmin)
+				precompile.SetContractNativeMinterStatus(state, noRoleAddr, precompile.AllowListEnabled)
```

Comment:
Can we add a separate address for the enabled status? We shouldn't use an address called `noRoleAddr` as an allow listed address.